### ISSUE TITLE
Don't set meta data on frozen objects

### ISSUE THREAD

**Issue by broerse** [emberjs/ember.js#16201]
To fix `Only string, number, symbol, boolean, null, undefined, and function are allowed as default properties (ember/avoid-leaking-state-in-ember-objects)` for ember 2.18 I changed my code to: 

```javascript
  postsSorting: Object.freeze(['date:desc']),
  arrangedContent: sort('posts', 'postsSorting'),
```

But then Ember stops working in IE 11. Perhaps Ember should not set meta data on frozen objects. The dutch warning I get `Kan de eigenschap __ember_meta__ niet definiëren: object is niet uitbreidbaar` that points to  `<-------------------------` 

But can't find it in https://github.com/emberjs/ember.js/blob/master/packages/ember-metal/lib/meta.js

Is it already fixed in Ember > 2.18 ?

```javascript
  var setMeta = void 0;
  exports.peekMeta = void 0;

  // choose the one appropriate for given platform
  if (emberUtils.HAS_NATIVE_WEAKMAP) {
    var getPrototypeOf = Object.getPrototypeOf;
    var metaStore = new WeakMap();

    setMeta = function WeakMap_setMeta(obj, meta) {
      {
        counters.setCalls++;
      }
      metaStore.set(obj, meta);
    };

    exports.peekMeta = function WeakMap_peekParentMeta(obj) {
      var pointer = obj;
      var meta = void 0;
      while (pointer !== undefined && pointer !== null) {
        meta = metaStore.get(pointer);
        // jshint loopfunc:true
        {
          counters.peekCalls++;
        }
        if (meta !== undefined) {
          return meta;
        }

        pointer = getPrototypeOf(pointer);
        {
          counters.peekPrototypeWalks++;
        }
      }
    };
  } else {
    setMeta = function Fallback_setMeta(obj, meta) {
      if (obj.__defineNonEnumerable) {
        obj.__defineNonEnumerable(EMBER_META_PROPERTY);
      } else {
        Object.defineProperty(obj, META_FIELD, META_DESC);   <-------------------------
      }

      obj[META_FIELD] = meta;
    };

    exports.peekMeta = function Fallback_peekMeta(obj) {
      return obj[META_FIELD];
    };
  }
```

**Comment by broerse** [comment]
Just tested and this bug is still present in 3.0.0-beta.6

**Comment by rwjblue** [comment]
Hmm, we don’t set meta as a property any more in 3.0+ (we always use a weakmap now).

**Comment by broerse** [comment]
It is exactly the same error I get in 3.0+ in IE 11.  You can see the error in https://bloggr.exmer.com/ after a refresh. The error is in dutch for me so it is an IE error. I believe the english translation is `Cannot define property:__ember_meta__, object is not extensible` and the error is not shown when I don't use the object freeze `postsSorting: Object.freeze(['date:desc']),`

Can I test something in 3.0+ ?

**Comment by broerse** [comment]
@rwjblue It is fixed in Ember v3.0.0. See https://bloggr.exmer.com/ in IE 11.

**Comment by rwjblue** [comment]
Thank you for testing and reporting back!
